# Problemas de prueba de hipótesis

## Ejercicio 1

Usted es un nutricionista que investiga dos tipos diferentes de dietas para ver si existe una diferencia significativa en la pérdida de peso después de un mes. Eliges dos grupos aleatorios de personas; un grupo sigue la primera dieta y el otro sigue la segunda. Al final del mes, se registra la pérdida de peso (en kg) de cada persona:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

Con estos datos se busca responder a la siguiente pregunta: ¿Existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta?

Para sacar conclusiones, siga los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba para comprobar la hipótesis. Puede utilizar la prueba t de Student.
-Analizar las conclusiones.

In [ ]:
from scipy.stats import ttest_ind

diet1 = [2.0, 2.5, 3.0, 2.8, 2.3, 2.7, 2.5]
diet2 = [3.0, 3.2, 3.1, 2.9, 2.8, 3.0, 3.2]

t_statistic, p_value = ttest_ind(diet1, diet2)
print(f"T-statistic: {t_statistic}, P-value: {p_value}")

alpha = 0.05

if p_value < alpha:
  print("Se rechaza la hipótesis nula (Ho), lo que indica que SÍ existe una diferencia significativa en la pérdida de peso promedio.")
else:
  print("No hay evidencia suficiente para rechazar la hipótesis nula (Ho), lo que indica que NO existe una diferencia significativa en la pérdida de peso promedio.")

T-statistic: -3.5383407969933938, P-value: 0.004083270191713912
Se rechaza la hipótesis nula (Ho), lo que indica que SÍ existe una diferencia significativa en la pérdida de peso promedio.


### CONCLUSION:
SÍ existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta.

## ANOVA

**ANOVA** (*Analysis of Variance - Análisis de variación*) Es una técnica estadística utilizada para comparar las medidas de dos o más grupos. La idea detrás de ANOVA es descomponer la variabilidad total de los datos en dos componentes: variabilidad entre grupos y variabilidad dentro de los grupos:

- **Variabilidad entre grupos**: Esta variabilidad se refiere a las diferencias entre las medias del grupo. Si esta variabilidad es considerablemente mayor que la variabilidad dentro del grupo, podría ser una indicación de que al menos una de las medias del grupo es diferente.
- **Variabilidad dentro del grupo**: Esta variabilidad se refiere a la dispersión de los datos dentro de cada grupo. Si todos los grupos tienen una variabilidad similar, entonces cualquier diferencia notable en las medias de los grupos podría considerarse significativa.

Las hipótesis en ANOVA suelen incluir:

- **Hipótesis nula** ($H₀$): Las medias de todos los grupos son iguales.
- **Hipótesis alternativa** ($H₁$): Al menos una de las medias del grupo es diferente.

Si el resultado de la prueba ANOVA es significativo (por ejemplo, un valor p inferior a un umbral como 0,05), esto sugiere que al menos la media de un grupo es diferente.

## Ejercicio 2

Un agricultor decide probar tres tipos diferentes de fertilizantes para determinar si alguno es superior en términos de producción de maíz. El agricultor planta maíz en 15 parcelas idénticas y utiliza los tres fertilizantes (5 parcelas para cada tipo). Al final de la temporada mide el rendimiento de maíz (en kg) de cada parcela, con el siguiente resultado:

| Fertilizante 1 | Fertilizante 2 | Fertilizante 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

Con estos datos busca responder la siguiente pregunta: ¿Existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes?

Para ayudarte, sigue los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba ANOVA.
-Analizar las conclusiones.
-Si un fertilizante es mejor que otro, ¿cómo podemos saberlo?

In [ ]:
from scipy.stats import f_oneway

Fertilizante1 = [20, 21, 20, 19, 20]
Fertilizante2 = [22, 21, 23, 22, 21]
Fertilizante3 = [24, 23, 22, 23, 24]

f_statistic, p_value = f_oneway(Fertilizante1, Fertilizante2, Fertilizante3)

print(f"F-statistic: {f_statistic}")
print(f"P-value: {p_value}")

alpha = 0.05

if p_value < alpha:
 print("Se rechaza la hipotesis nula (Ho), lo que indica que SÍ existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes.")
else:
 print("No hay evidencia suficiente de que exista una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes.")

F-statistic: 20.31578947368421
P-value: 0.000140478247931904
Se rechaza la hipotesis nula (Ho), lo que indica que SÍ existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes.


### CONCLUSION:
SÍ existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes.

In [ ]:
# PARA SABER SI UN FERTILIZANTE ES MEJOR QUE OTRO, SE REALIZARA USANDO UNA FUNCION
# POST HOC DE SCIPY, LLAMADA tukey_hsd (Honestly Significant Difference).

# ESTA FUNCION SE UTILIZA PARA COMPARAR PARES DE MEDIAS,
# DESPUES DE HABER HECHO UN ANOVA (Análisis de Varianza)

from scipy.stats import tukey_hsd

Fertilizante1 = [20, 21, 20, 19, 20]
Fertilizante2 = [22, 21, 23, 22, 21]
Fertilizante3 = [24, 23, 22, 23, 24]

mejor_fertilizante = tukey_hsd(Fertilizante1, Fertilizante2, Fertilizante3)
print(mejor_fertilizante)

Pairwise Group Comparisons (95.0% Confidence Interval)
Comparison  Statistic  p-value  Lower CI  Upper CI
 (0 - 1)     -1.800     0.010    -3.143    -0.457
 (0 - 2)     -3.200     0.000    -4.543    -1.857
 (1 - 0)      1.800     0.010     0.457     3.143
 (1 - 2)     -1.400     0.041    -2.743    -0.057
 (2 - 0)      3.200     0.000     1.857     4.543
 (2 - 1)      1.400     0.041     0.057     2.743



GRUPO 0 = Fertilizante1

GRUPO 1 = Fertilizante2

GRUPO 2 = Fertilizante3

# CONCLUSION:

Fertilizante 1 vs. Fertilizante 2 (0 - 1): El valor p es 0.010, que es menor
que 0.05. Esto indica una diferencia significativa en el rendimiento promedio entre el Fertilizante 1 y el Fertilizante 2.

Fertilizante 1 vs. Fertilizante 3 (0 - 2): El valor p es 0.000, que es menor que 0.05. Esto indica una diferencia muy significativa en el rendimiento promedio entre el Fertilizante 1 y el Fertilizante 3.

Fertilizante 2 vs. Fertilizante 3 (1 - 2): El valor p es 0.041, que es menor que 0.05. Esto indica una diferencia significativa en el rendimiento promedio entre el Fertilizante 2 y el Fertilizante 3.

Esto significa que el Fertilizante 3 es el que ofrece el mejor rendimiento, despues el Fertilizante 2, y el Fertilizante 1 es el que menor rendimiento tiene.